paho/client.go

```diff
@@ -321,7 +355,7 @@ func (c *Client) Connect(ctx context.Context, cp *Connect) (*Connack, error) {
 		c.workers.Add(1)
 		go func() {
 			defer c.workers.Done()
-			defer c.debug.Println("returning from ack tracker routine")
+			defer c.debug.Println("ack tracker worker returned")
```

Comment:
This function will end up being a race (because sending the queued acks requires that `writeLoop()` is running but that also stops when `c.stop` is closed).